src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -208,5 +227,130 @@ static void VerifyMemberOnlyApplyToTypesOrStrings (SymbolAnalysisContext context
+							method.GetDisplayName (), overriddenMethod.GetDisplayName ()));
+					}
+				}
+				basestTypeSymbol = basestTypeSymbol.BaseType;
+			}
+		}
+
+		static void VerifyDamOnInterfaceAndImplementationMethodsMatch (INamedTypeSymbol type, SymbolAnalysisContext context)
+		{
+			foreach (var (interfaceMember, implementationMember) in type.GetMemberInterfaceImplementationPairs ()) {
+				if (implementationMember is IMethodSymbol implementationMethod
+					&& interfaceMember is IMethodSymbol interfaceMethod)
+					VerifyMethodDamAttributesMatch (implementationMethod, interfaceMethod, context);
+			}
+		}
+
+		static void VerifyDamOnPropertyAndAccessorMatch (IMethodSymbol methodSymbol, SymbolAnalysisContext context)
+		{
+			if (!(methodSymbol.AssociatedSymbol?.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var propertyAttribute) == true)
+				|| (propertyAttribute.ConstructorArguments.Length == 1 && propertyAttribute.ConstructorArguments[0].Value is (int)DynamicallyAccessedMemberTypes.None))
+				return;
+
+			if (methodSymbol.MethodKind == MethodKind.PropertyGet
+				&& methodSymbol.TryGetReturnTypeAttribute (DynamicallyAccessedMembersAttribute, out var getValueAttribute)
+				&& !(getValueAttribute.ConstructorArguments.Length == 1 && getValueAttribute.ConstructorArguments[0].Value is (int)DynamicallyAccessedMemberTypes.None)) {
+				context.ReportDiagnostic (Diagnostic.Create (
+					DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersConflictsBetweenPropertyAndAccessor),
+					methodSymbol.AssociatedSymbol!.Locations[0],
+					methodSymbol.AssociatedSymbol!.GetDisplayName (),
+					methodSymbol.GetDisplayName ()
+				));
+				return;
+			}
+
+			if (methodSymbol.MethodKind == MethodKind.PropertySet
+				&& methodSymbol.Parameters[0].TryGetAttribute (DynamicallyAccessedMembersAttribute, out var setValueAttribute)
+				&& !(setValueAttribute.ConstructorArguments.Length == 1 && setValueAttribute.ConstructorArguments[0].Value is (int)DynamicallyAccessedMemberTypes.None)) {
+				context.ReportDiagnostic (Diagnostic.Create (
+					DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersConflictsBetweenPropertyAndAccessor),
+					methodSymbol.AssociatedSymbol!.Locations[0],
+					methodSymbol.AssociatedSymbol!.GetDisplayName (),
+					methodSymbol.GetDisplayName ()
+				));
+				return;
+			}
+		}
+
+		// DAM flows from property to ".get" return value
+		static bool TryGetReturnDamAttribute (IMethodSymbol methodSymbol, [NotNullWhen (true)] out AttributeData? attribute)
+		{
+			// Accessor is used when both are present. Warn on conflicts with VerifyDamOnPropertyAndAccessorMatch
+			return methodSymbol.TryGetReturnTypeAttribute (DynamicallyAccessedMembersAttribute, out attribute)
+				? true
+				: methodSymbol.MethodKind == MethodKind.PropertyGet
+					&& methodSymbol.AssociatedSymbol!.TryGetAttribute (DynamicallyAccessedMembersAttribute, out attribute);
+		}
```

Comment:
This should already be handled by `FlowAnnotations.GetMethodReturnValueAnnotation` - and the one for parameters is done in `FlowAnnotations.GetMethodReturnValueAnnotation`. They also extract the annotation value from the attribute and so on.